include/envoy/http/filter.h

```diff
@@ -418,6 +418,20 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
   // Note that HttpConnectionManager sanitization will *not* be performed on the
   // recreated stream, as it is assumed that sanitization has already been done.
   virtual bool recreateStream() PURE;
+
+  /**
+   * Adds socket options to be applied to the upstream request. Note that unique values for the
+   * options will likely lead to many connection pools being created. The added options are appended
+   * to any previously added.
```

Comment:
They are also appended to the downstream options if present, correct?

Reply:
Yup. I'll mention that.

Reply:
Note that they are appended in a different spot -- i.e. during connection pool creation, rather than immediately. But, the effect ends up being the same. I'll make the note here, but it does assume that the connection pool creation behaviour won't change.